src/bp/core/services/telemetry/telemetry-stats.ts

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios'
+import LicensingService from 'common/licensing-service'
+import Database from 'core/database'
+import { TelemetryRepository } from 'core/repositories/telemetry_payload'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import yn from 'yn'
+
+import { GhostService } from '..'
+import { JobService } from '../job-service'
+
+const debug = DEBUG('stats')
+const TELEMETRY_URL = process.TELEMETRY_URL
+
+@injectable()
+export abstract class TelemetryStats {
+  protected abstract url: string
+  protected abstract lock: string
+  protected abstract interval: number
+
+  constructor(
+    @inject(TYPES.GhostService) protected ghostService: GhostService,
+    @inject(TYPES.Database) private database: Database,
+    @inject(TYPES.LicensingService) protected licenseService: LicensingService,
+    @inject(TYPES.JobService) private jobService: JobService,
+    @inject(TYPES.TelemetryRepository) private telemetryRepo: TelemetryRepository
+  ) {}
+
+  public async start() {
+    await this.run(this.getStats.bind(this), this.lock, this.interval, this.url)
+
+    setInterval(this.run.bind(this, this.getStats.bind(this), this.lock, this.interval, this.url), this.interval)
+  }
+
+  protected abstract getStats()
+
+  protected async run(job: Function, lockResource: string, interval: number, url: string) {
+    const lock = await this.jobService.acquireLock(lockResource, interval - ms('1 minute'))
+    if (lock) {
+      debug('Acquired lock')
+      const stats = await job()
+      await this.sendStats(url, stats)
+    }
+  }
```

Comment:
If you only have one job, you can simplify this and remove the job parameter

Reply:
fixed in 56f1ff721acee12444bd695cb0c065a45fe254f7